### DISCUSSION TITLE
What are the detailed version differences from jquey1.7.2 to 1.12.4?  I made a lot of mistakes with jquery

### DISCUSSION THREAD

**Discussion by zhaomaolin264** [jquery/jquery#5026] [Q&A] [answered]
What are the detailed version differences from jquey1.7.2 to 1.12.4?  I made a lot of mistakes with jquery

**Comment by mgol** [comment, marked as answer]
You can read blog posts of all the minor releases in between:
* https://blog.jquery.com/2012/08/09/jquery-1-8-released/
* https://blog.jquery.com/2013/01/15/jquery-1-9-final-jquery-2-0-beta-migrate-final-released/
* https://blog.jquery.com/2013/05/24/jquery-1-10-0-and-2-0-1-released/
* https://blog.jquery.com/2014/01/24/jquery-1-11-and-2-1-released/
* https://blog.jquery.com/2016/01/08/jquery-2-2-and-1-12-released/

That said, most of the changes were non-breaking with the exception of 1.9 that had many breaking changes. Reading the following upgrade guide is recommended: https://jquery.com/upgrade-guide/1.9/.

It's also a good idea to update jQuery to 1.12.4 and add jQuery Migrate 1.4.1 and then gradually fix reported violations. When warnings are gone, you should remove jQuery Migrate.
